streams/shared/src/main/scala/scalaz/stream/SinkPure.scala

```diff
@@ -21,33 +21,28 @@ import scalaz.zio._
 trait SinkPure[+E, +A0, -A, +B] extends ZSink[Any, E, A0, A, B] { self =>
   import ZSink.Step
 
-  override def initial              = IO.succeed(initialPure)
-  override def step(s: State, a: A) = IO.succeed(stepPure(s, a))
-  override def extract(s: State)    = IO.fromEither(extractPure(s))
-
-  def initialPure: Step[State, Nothing]
-  def stepPure(s: State, a: A): Step[State, A0]
-  def extractPure(s: State): Either[E, B]
-
-  def stepChunkPure[A1 <: A](s: State, as: Chunk[A1]): Step[State, A0] = {
-    val len = as.length
-
-    def loop(s: Step[State, A0], i: Int): Step[State, A0] =
-      if (i >= len) s
-      else if (Step.cont(s)) loop(stepPure(Step.state(s), as(i)), i + 1)
-      else s
-
-    loop(Step.more(s), 0)
-  }
+  override def contramap[C](f: C => A): SinkPure[E, A0, C, B] =
```

Comment:
I went through this one but now realise `SinkPure` should probably be package-private, like `StreamPure`?